src/main/java/uk/gov/hmcts/reform/cpo/security/SecurityUtils.java

```diff
@@ -37,4 +46,12 @@ public String getServiceNameFromS2SToken(String serviceAuthenticationToken) {
     private String removeBearerFromToken(String token) {
         return token.startsWith(BEARER) ? token.substring(BEARER.length()) : token;
     }
+
+    public boolean hasFullAccessToService(String token) {
+        if (isBlank(token)) {
```

Comment:
I guess we might need fine grained with in "CRUD" itself?. Like some services have "C" permission (create) only.

Reply:
That's correct. We need to be able for each service to specify it's fully CRUD permissions. So we need something more fine grained than what's currently being implemented. We could define the following (pseudocode):

enum CrudPermission {
    C,R,U,D
}

```
class ServicePermissions {
  String id;   // service id
  String crudPermissions; // the permissions as a string, e.g. "CR" or "U" or "DUCR"
  
  public boolean hasPermission(Set<CrudPermission> permissions) {
  // true if all permission passed as arguments are contained in this.permissions
  }
}
 Then in the config file :
 s2s:
    permissions:
         - 
           id: payment_app
           crudPermissions: R
         - 
          id:  anotherService
          crudPermissions: DR
....
     
spring should be able to parse the config into a List<ServicePermissions>

```

Reply:
👍

Reply:
Just to confirm, an endpoint open to anyone with Create, Update and Read permissions would be defined as follows

`@PreAuthorize("@securityUtils.hasCreate() && @securityUtils.hasUpdate() && @securityUtils.hasRead()")`

Reply:
and perhaps a 

`@PreAuthorize("@securityUtils.hasCRUD()`

Reply:
if that's gonna happen in the future this will do as you suggested:
```
@PreAuthorize("@securityUtils.hasCRUD()
```